### DISCUSSION TITLE
Why does `vscode.window.showTextDocument(vscode.workspace.openTextDocument())` work without the async keyword?

### DISCUSSION THREAD

**Discussion by phommy** [microsoft/vscode-discussions#1106] [Extension Development QnA] [answered]
I've noticed that a piece of code I've been using,
```javascript
vscode.window.showTextDocument(vscode.workspace.openTextDocument());
```
to create an Untitled document, instead of
```javascript
vscode.window.showTextDocument(async vscode.workspace.openTextDocument());
```
has been running correctly. Although the declaration for `showTextDocument` is
```typescript
export function showTextDocument(document: TextDocument, column?: ViewColumn, preserveFocus?: boolean): Thenable<TextEditor>;
```
it seems like this isn't a mechanism of async/await itself, but rather that `showTextDocument` has some sort of **fault tolerance** built in.

Could you please help me:
1. I've downloaded the VSCode source code (using `git clone https://github.com/microsoft/vscode.git`) and want to examine the implementation of `vscode.window.showTextDocument`, but after using grep, I've excluded all the locations I found. Could you point me to the file where the method is implemented?
2. If I'm providing an API to others, should I also implement this kind of fault tolerance?

**Comment by isidorn** [comment, marked as answer]
For 1. I suggest using Copilot with @workspace over the VS Code code base
2. Most likely no. But up to you as the owner of your API.
